src/Mailer/Transport/SmtpTransport.php

```diff
@@ -394,6 +397,26 @@ protected function _authLogin(string $username, string $password): void
         }
     }
 
+    /**
+     * Authenticate using AUTH XOAUTH2 mechanism.
+     *
+     * @param string $username Username.
+     * @param string $token Token.
+     * @return void
+     * @see https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#smtp-protocol-exchange
+     * @see https://developers.google.com/gmail/imap/xoauth2-protocol#smtp_protocol_exchange
+     */
+    protected function _authXoauth2(string $username, string $token): void
+    {
+        $authString = base64_encode(sprintf(
+            "user=%s\1auth=Bearer %s\1\1",
```

Comment:
I have to wonder what led to the choice to use `^A` as the separator :thinking:

Reply:
Yeah, without some background info that does seems like a pretty strange character choice.